### ISSUE TITLE
False positive on archive.org

### ISSUE THREAD

**Issue by philippelatulippe** [Sayrus/audiocontext-scrambler#2]
Hello! First off all thank you for creating this extension. I was a user of AFD but something seems very off about their website, not to mention that the way it injects scripts breaks certain sites and non-html documents.

Just like AFD, the extension detect a fingerprinting attempt when I use the emscripten-based dosbox emulator on archive.org. The emulator supports sound, so it's expected that it would use the audiocontext api, but I doubt archive.org would be trying to fingerprint my browser. Hopefully there's a way to distinguish between what the emulator is doing and malicious fingerprinting attempts.

Example:
https://archive.org/details/msdos_Fast_Tracker_2_1994

On top of this, if I enable notifications, the extension constantly generates extensions when visiting that page, eventually crashing firefox. If this is a separate issue I can open another ticket.

**Comment by Sayrus** [comment]
Hello,

Thanks for the feedback !
I should be able to easily patch the notification to display only once per URL, domain or injection. However, I don't think I can distinguish between legitimate and malicious uses of the AnalyzerNode.
If I can't distinguish them, then the best solution is to introduce a whitelist that is both on Github and customizable by the user.

I will investigate the application you linked to see if there might be a way to distinguish uses.

**Comment by Sayrus** [comment]
It looks like the best way to do this is to have the background service handles everything and use:
- `onBeforeRequest` to check if the URL matches against our allowlist/blocklist
-  `browser.tabs.executeScript` to inject the content script

This should also give more control over what is injected as we might be able to use some external variables without having to wait for an async call.
